ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+      A patch encoding the change between the data in the event
+      property, and the data after the update.
+
+   If the change only affects a single instance of a recurring event,
+   the server MAY set the event and eventPatch properties for just that
+   instance; the calendarEventId MUST still be for the base event.
+
+7.3.  CalendarEventNotification/get
+
+   This is a standard "/get" method as described in Section 5.1 of
+   [RFC8620].
+
+7.4.  CalendarEventNotification/changes
+
+   This is a standard "/changes" method as described in Section 5.2 of
+   [RFC8620].
+
+7.5.  CalendarEventNotification/set
+
+   This is a standard "/set" method as described in Section 5.3 of
+   [RFC8620].
+
+   Only destroy is supported; any attempt to create/update MUST be
+   rejected with a forbidden SetError.
+
+7.6.  CalendarEventNotification/query
+
+   This is a standard "/query" method as described in Section 5.5 of
+   [RFC8620].
+
+7.6.1.  Filtering
+
+   A *FilterCondition* object has the following properties:
+
+   *  *after*: UTCDateTime|null
+      The creation date must be on or after this date to match the
+      condition.
+   *  *before*: UTCDateTime|null
+      The creation date must be before this date to match the condition.
+   *  *type*: String
+      The type property must be the same to match the condition.
+   *  *calendarEventIds*: Id[]|null
+      A list of event ids.  The calendarEventId property of the
+      notification must be in this list to match the condition.
+
+
+
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 40]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+7.6.2.  Sorting
+
+   The "created" property MUST be supported for sorting.
+
+7.7.  CalendarEventNotification/queryChanges
+
+   This is a standard "/queryChanges" method as described in Section 5.6
+   of [RFC8620].
+
+8.  Examples
+
+   For brevity, in the following examples only the "methodCalls"
+   property of the Request object, and the "methodResponses" property of
+   the Response object is shown.
+
+8.1.  Fetching initial data
+
+   A user has authenticated and the client has fetched the JMAP Session
+   object.  It finds a single Account with the
+   urn:ietf:params:jmap:calendars capability, with id "a0x9", and wants
+   to display all the calendar information for January 2023 in the
+   Australia/Melbourne time zone.  It might make the following request:
+
+   [
+     ["Calendar/get", {
+       "accountId": "a0x9"
+     }, "0"],
+     ["ParticipantIdentity/get", {
+       "accountId": "a0x9"
+     }, "1"],
+     ["CalendarEvent/query", {
+       "accountId": "a0x9",
+       "timeZone": "Australia/Melbourne",
+       "filter": {
+         "after": "2023-01-01T00:00:00",
+         "before": "2023-02-01T00:00:00"
+       }
+     }, "2"],
+     ["CalendarEvent/get", {
+           "accountId": "a0x9",
```

Comment:
Indenting here doesn't match the rest of the example

Reply:
Thanks, fixed.